### PR TITLE
[#777] Use GH latest release version

### PR DESCRIPTION
**Pull request by kamranayub**
Related #777

## Changes:

- On local builds, use local package version placeholder for quick builds
- On CI builds, use GitHub release version. Fails the build if API request fails.

This change allows us to remove all specific versioning from the repository, relying on GitHub as the source of truth for all releases. As a side effect, dist files get a better version anyway--previously they only had the latest release (v0.10.0) and date. Now they will get the commit and Travis build number.

### CODE REVIEW

**Comment by kamranayub** [comment]
Merge #782 and then I can merge that into this PR to fix build.

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "excalibur",
   "private": true,
   "description": "Excalibur.js is a simple JavaScript game engine with TypeScript bindings for making 2D games in HTML5 Canvas. Our mission is to make web game development as simple as possible.",
-  "version": "0.10.0",
+  "version": "0.0.1",
```
Comment (eonarheim) [inline package.json]: Why did we back the verion up to 0.0.1?

**Comment by kamranayub** [comment]
Because this isn't a real version and I wanted to make that clear. Local
builds will use this only, production builds use GH release.

**Comment by kamranayub** [comment]
The only place this will be confusing is for people browsing the repo. If
there's a clearer semver that communicates this is a fake version, I'm up
for it. I think npm requires a version, hence why I didn't delete the
property otherwise I would've.

**Comment by eonarheim** [comment]
@kamranayub can we do a piece of non-valid semver then? like `0.0.local` or something?

**Comment by kamranayub** [comment]
Maybe, I will check. I thought npm would fail if it couldn't parse
version...

**Comment by kamranayub** [comment]
This is ready. Appveyor PR build fails due to missing token, but will be there once merged.

    Error: Missing environment variable GH_TOKEN

**Pull request merged**